CRM/Financial/BAO/Payment.php

```diff
@@ -80,10 +79,7 @@ public static function create($params) {
     }
     if (!$fullyPaidPayLater) {
       $trxn = CRM_Contribute_BAO_Contribution::recordPartialPayment($contribution, $params);
-      $paid = CRM_Core_BAO_FinancialTrxn::getTotalPayments($params['contribution_id']);
-      $total = CRM_Core_DAO::getFieldValue('CRM_Contribute_DAO_Contribution', $params['contribution_id'], 'total_amount');
-      $cmp = bccomp($total, $paid, 5);
-      if ($cmp == 0 || $cmp == -1) {// If paid amount is greater or equal to total amount
```

Comment:
I think we shouldn't replace this code because, for partial payment or refund, the balance needs to be recalculated and compared against contribution's total_amount to see if the payments actually complete the contribution after calling```recordPartialPayment()```. Current patch only satisfies 'Pending' payment. What do you think @eileenmcnaughton ?

Reply:
@monishdeb so per above - I figure if we get the balance BEFORE adding the payment we should be able to figure out what impact the payment has on the balance